desc/compute/_field.py

```diff
@@ -2286,12 +2306,55 @@ def _B_mag_t(params, transforms, profiles, data, **kwargs):
     ],
 )
 def _B_mag_z(params, transforms, profiles, data, **kwargs):
-    data["|B|_z"] = (
+    data["|B|_z"] = safediv(
         data["B^theta_z"] * data["B_theta"]
         + data["B^theta"] * data["B_theta_z"]
         + data["B^zeta_z"] * data["B_zeta"]
-        + data["B^zeta"] * data["B_zeta_z"]
-    ) / (2 * data["|B|"])
+        + data["B^zeta"] * data["B_zeta_z"],
+        2 * data["|B|"],
+    )
+    return data
+
+
+@register_compute_fun(
+    name="|B|_alpha",
+    label="\\partial_{\\alpha} |\\mathbf{B}|",
+    units="T",
+    units_long="Tesla",
+    description="Magnitude of magnetic field, derivative wrt field line angle",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="rtz",
+    data=["|B|_t", "alpha_t"],
+)
+def _B_mag_alpha(params, transforms, profiles, data, **kwargs):
+    # constant ρ and ζ
+    data["|B|_alpha"] = safediv(data["|B|_t"], data["alpha_t"])
+    return data
+
+
+@register_compute_fun(
+    # TODO: pick a name
```

Comment:
maybe something like ``|B|_z|r,a``? or ``|B|_l`` since its along a field line (though technically not derivative wrt length)